platform/solr/solr-factory/src/main/java/org/codice/solr/factory/SolrServerFactory.java

```diff
@@ -164,7 +164,7 @@ static SolrServer getHttpSolrServer() {
         SolrServer server;
         try {
             server = getSolrServer(url, coreName, configFile, coreUrl);
-        } catch (SolrException ex) {
+        } catch (Exception ex) {
```

Comment:
It is intentional, though I could be talked out of it. The problem is that some of the edge cases we want to handle here throw unchecked exceptions. I found and fixed some manually, but by then you have a catch clause with four or five exceptions, and you still can't be sure that's all of them. I couldn't really think of any exceptions where we would want to *not* retry, so I think this logic is valid.

There is logging further down in this class. We could log here as well, but in many cases it would result in double logging. I don't mind that so much, but I think we were trying to reduce it at some point. I remain undecided.